dotCMS/src/main/java/com/dotcms/browser/BrowserAPI.java

```diff
@@ -0,0 +1,583 @@
+	 * 						If false, returns the live version of HTMLPages, Links and FileAssets in the folder.
+	 *
+	 * @param showArchived  If true, includes the archived version of HTMLPages, Links and FileAssets in the folder.
+	 *
+	 * @param noFolders
+	 * @param onlyFiles
+	 * @param sortBy
+	 * @param sortByDesc
+	 * @param excludeLinks
+	 * @return
+	 * @throws DotSecurityException
+	 * @throws DotDataException
+	 */
+    @Deprecated
+	public Map<String, Object> getFolderContent(User user, String folderId,
+			int offset, int maxResults, String filter, List<String> mimeTypes,
+			List<String> extensions, boolean showWorking, boolean showArchived, boolean noFolders,
+			boolean onlyFiles, String sortBy, boolean sortByDesc, boolean excludeLinks, long languageId, boolean dotAssets)
+			throws DotSecurityException, DotDataException {
+
+	    final boolean showPages=!onlyFiles;
+
+	    BrowserQuery browserQuery = BrowserQuery.builder()
+	                    .showDotAssets(dotAssets)
+	                    .showLinks(!excludeLinks)
+	                    .showExtensions(extensions)
+	                    .withFilter(filter)
+	                    .withHostOrFolderId(folderId)
+	                    .withLanguageId(languageId)
+	                    .offset(offset)
+	                    .showFiles(true)
+	                    .showPages(showPages)
+	                    .showFolders(!noFolders)
+	                    .showArchived(showArchived)
+	                    .showWorking(showWorking)
+	                    .sortBy(sortBy)
+	                    .sortByDesc(sortByDesc)
+	                    .withUser(user).build();
+	    
+	    return getFolderContent(browserQuery);
+
+	}
+	
+    @CloseDBIfOpened
+    public Map<String, Object> getFolderContent(final BrowserQuery browserQuery) throws DotSecurityException, DotDataException {
+
+
+        List<Map<String, Object>> returnList = new ArrayList<>();
+
+        Role[] roles = APILocator.getRoleAPI().loadRolesForUser(browserQuery.user.getUserId())
+                        .toArray(new Role[0]);
+
+
+        // gets folder parent
+        Folder parent = folderAPI.find(browserQuery.hostFolderId, browserQuery.user, false);
+
+
+        Host host = null;
+        if (parent == null) {// If we didn't find a parent folder lets verify if
+                             // this is a host
+            host = APILocator.getHostAPI().find(browserQuery.hostFolderId, browserQuery.user, false);
+
+            if (host == null) {
+                Logger.error(this, "Folder ID doesn't belong to a Folder nor a Host, id: " + browserQuery.hostFolderId
+                                + ", maybe the Folder was modified in the background.");
+                throw new NotFoundInDbException("Folder ID doesn't belong to a Folder nor a Host, id: " + browserQuery.hostFolderId);
+            }
+        }
+
+        if (browserQuery.showFolders) {
+            if (parent != null) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)